src/java_bytecode/generate_java_generic_type.cpp

```diff
@@ -216,3 +223,38 @@ irep_idt generate_java_generic_typet::build_generic_tag(
 
   return new_tag_buffer.str();
 }
+
+/// Build the specalised version of the specific class, with the specified
+/// parameters and name.
+/// \param generic_class_definition: The generic class we are specialising
+/// \param new_tag: The new name for the class (like Generic<java::Float>)
+/// \param new_components: The specialised components
+/// \return The newly constructed class.
+java_class_typet
+generate_java_generic_typet::construct_specialised_generic_type(
+  const java_generic_class_typet &generic_class_definition,
+  const irep_idt &new_tag,
+  const struct_typet::componentst &new_components) const
+{
+  java_class_typet specialised_class = generic_class_definition;
+  // We are specialising the logic - so we don't want to be marked as generic
+  specialised_class.set(ID_C_java_generics_class_type, false);
```

Comment:
Consider providing methods (`generic_class_type()` or similar) that do this?
(Same for all manual .set and .get usage)

Reply:
This is a temporary work around to unblock generics - @NlightNFotis is working on a derivation of `java_class_typet` (`java_specialised_generic_class_typet`) which should tidy this logic away.